Laravel 11 support

ifox opened the pull request:
Very excited to share that Twill 3 has been successfully tested with Laravel 11! Read about [Laravel 11 on Laravel News](https://laravel-news.com/laravel-11).

We will need these PRs on our dependencies to be merged and released:

- https://github.com/Astrotomic/laravel-translatable/pull/387
- https://github.com/lazychaser/laravel-nestedset/pull/594
- https://github.com/cartalyst/tags/pull/47
- https://github.com/spatie/laravel-activitylog/pull/1276
- https://github.com/spatie/laravel-analytics/pull/514

This PR can be tested using the following Composer configuration:

```json
"repositories": [
    {
        "type": "vcs",
        "url": "git@github.com:area17/laravel-translatable.git"
    },
    {
        "type": "vcs",
        "url": "git@github.com:area17/laravel-nestedset.git"
    },
    {
        "type": "vcs",
        "url": "git@github.com:ifox/tags.git"
    },
    {
        "type": "vcs",
        "url": "git@github.com:StevePorter92/laravel-activitylog.git"
    }
],
"require": {
    "php": "^8.2",
    "area17/twill": "dev-laravel-11-support",
    "laravel/framework": "^11.0",
    "laravel/tinker": "^2.9"
},
```

The main change in Laravel 11 is the streamlined skeleton. When creating a fresh Laravel 11 application, your repository will contains 69 less files than it would on Laravel 10! Read more about this on [Laravel News](https://laravel-news.com/laravel-11-directory-structure). 

I'm personally very excited by the new config directory behavior, which matches the behavior of Twill's configuration file, where you only need to override what you need, and everything else just inherits from the provided defaults.

It looks so good!
```
app
├── Http
│   ├── Controllers
│   │   └── Twill
│   └── Requests
│       └── Twill
├── Models
├── Providers
│   └── AppServiceProvider.php
├── Repositories
└── View
    └── Components
        └── Twill

bootstrap
├── cache
├── app.php
└── providers.php

config
├── translatable.php
└── twill.php

routes
├── console.php
├── twill.php
└── web.php
```

ifox committed:
Add Laravel 11 support

ifox committed:
Merge branch '3.x' into laravel-11-support

driftingly commented:
- https://github.com/Astrotomic/laravel-translatable/pull/391
- https://github.com/lazychaser/laravel-nestedset/pull/596 - not yet tagged
- https://github.com/cartalyst/tags/commit/df34ce8093d83918777d8409fbdecf7de05c7b28 - released in v14

kfina-planxy commented:
I have problems installing this for laravel 11:

my composer.json
```
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/area17/twill.git",
            "canonical": false
        },
        {
            "type": "vcs",
            "url": "git@github.com:area17/laravel-translatable.git",
            "canonical": false
        },
        {
            "type": "vcs",
            "url": "git@github.com:area17/laravel-nestedset.git",
            "canonical": false
        },
        {
            "type": "vcs",
            "url": "git@github.com:ifox/tags.git",
            "canonical": false
        },
        {
            "type": "vcs",
            "url": "git@github.com:StevePorter92/laravel-activitylog.git",
            "canonical": false
        }
    ],
    "require": {
        "php": "^8.2.0",
        "area17/twill": "dev-laravel-11-support",
        "barryvdh/laravel-dompdf": "^2.0",
        "guzzlehttp/guzzle": "^7.2",
        "inertiajs/inertia-laravel": "^1.0",
        "mongodb/laravel-mongodb": "^4.1.1",
        "laravel-notification-channels/microsoft-teams": "^1.1",
        "laravel/framework": "^11.0",
        "laravel/jetstream": "^5.0",
        "laravel/sanctum": "^4.0",
        "laravel/tinker": "^2.9",
        "orangehill/iseed": "^3.0",
        "spatie/laravel-googletagmanager": "^2.6",
        "spatie/laravel-permission": "^6.0",
        "tightenco/ziggy": "^1.0"
    },
```
And get the following error

Problem 1
    - Root composer.json requires laravel/framework ^11.0 -> satisfiable by laravel/framework[v11.0.0, ..., v11.3.0].
    - area17/twill dev-laravel-11-support requires astrotomic/laravel-translatable ^v11.12 || dev-laravel-11-support -> satisfiable by astrotomic/laravel-translatable[v11.12.0, v11.12.1].
    - Conclusion: don't install astrotomic/laravel-translatable v11.12.1 (conflict analysis result)
    - Root composer.json requires area17/twill dev-laravel-11-support -> satisfiable by area17/twill[dev-laravel-11-support].

**Note:**
I had to add the canonical: false, because otherwise it didn't want to install because there was a "higher priority" package available.

driftingly commented:
@ifox @kfina-planxy, it looks like the remaining packages have been updated 🥳 
https://github.com/Astrotomic/laravel-translatable/releases/tag/v11.13.0

ifox committed:
Update composer dependencies to stable versions supporting Laravel 11

ifox committed:
Add support for php-invoker > 4

ifox committed:
Upgrade rector

ifox committed:
Remove torchlight-commonmark docs build only dependency

ifox committed:
Add CACHE_STORE to phpunit variables for Laravel 11 support

ifox committed:
Fix rector breaking change

ifox committed:
Fix phpunit breaking change

ifox committed:
Fix phpunit breaking change

ifox committed:
Add testbench:migrate before building to prevent cache table missing

This is because testbench now defaults to the database cache store instead of file, see https://github.com/orchestral/testbench-core/commit/1bb8919e0b118c029a483cace2dda18d6749f2f3#diff-86a4a158399429a6fc000a17b3a8da05fddb1c0c16aab3d8ef00f6c67cb8154dR28.

Even though our phpunit is configured to use a mysql database, we need that temporary testing database to avoid failing tests because the cache table doesn't exist during the Twill assets build.

ifox committed:
Remove Laravel 9 testing with PHP 8.0 from matrix

ifox committed:
Attempt to force array cache store during build

ifox committed:
Set default twill user model in browser test case

This shouldn't be necessary as this should be coming from Twill's default configuration that is merged in Twill's service provider. This wasn't an issue with previous versions. Not sure yet if the issue is coming from Laravel 11, Dusk or Testbench.

ifox committed:
Add noSandbox to chrome driver options in headless mode

ifox committed:
Manually start chrome driver in CI

For some reason this isn't needed prior to Laravel 11. Still trying to figure out what has changed in the ecosystem that broke our Dusk tests.

ifox committed:
Migrate phpunit configuration

Co-Authored-By: Antonio Ribeiro <acr@antoniocarlosribeiro.com>

ifox committed:
Move data providers to be static methods

Co-Authored-By: Antonio Ribeiro <acr@antoniocarlosribeiro.com>

ifox committed:
Revert "Set default twill user model in browser test case"

This reverts commit a3a3337fe13a9697d639e809d47c03fbd412a980.

ifox committed:
Fix skeleton base path for Dusk

ifox committed:
Add back CI default db password

ifox committed:
Disable example module provider in browser test case

ifox committed:
Revert "Disable example module provider in browser test case"

This reverts commit 05d92d37c3eef31f9e7b93bb7c231925bb1a3127.

ifox committed:
Manually register correct path for testbench-core / testbench-dusk apps

ifox committed:
Enable npm install during twill:build in CI test

ifox committed:
Enable phpunit stop-on-error in CI

ifox committed:
Register namespace after installing preset files

ifox committed:
Yet another attempt to get Testbench to behave as needed in CI

ifox committed:
Ensure testbench-dusk discovers packages before running tests

This is so that the vendor symlink is correctly created in Dusk's Laravel skeleton

ifox committed:
Bring back L9/P8 testing

Patch test case base classes in CI to deal with PHPUnit breaking changes

ifox committed:
Fix phpunit configuration warnings

ifox committed:
Remove unused import
